### PR TITLE
add youdao translate

### PR DESCRIPTION
**Pull request by tiancheng92**
## Description

This extension brings Youdao translate to raycast!

## Type of change

- New extension

## Screencast

form
result
select_form

## Checklist

- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/youdao-translate/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "youdao-translate",
+  "title": "Youdao Translate",
+  "description": "Translate Content By Youdao API",
```
Comment (PitNikola) [inline extensions/youdao-translate/package.json:4]:
```suggestion
  "description": "Translate text using Youdao",
```

File: extensions/youdao-translate/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "icon": "translate.png",
+  "author": "tiancheng92",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Youdao Translate",
```
Comment (PitNikola) [inline extensions/youdao-translate/package.json:11]:
```suggestion
      "title": "Translate",
```

File: extensions/youdao-translate/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      "subtitle": "Translate Content",
```
Comment (PitNikola) [inline extensions/youdao-translate/package.json:12]:
```suggestion
      "subtitle": "Youdao",
```

File: extensions/youdao-translate/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      "description": "Translate Content By Youdao API",
```
Comment (PitNikola) [inline extensions/youdao-translate/package.json:13]:
```suggestion
      "description": "Translate text using Youdao",
```

**Review by PitNikola — CHANGES_REQUESTED** [review]
Hey @tiancheng92. Thanks for your contribution!
Do you think you could provide me with API key so I could test your extension? Feel free to send me it via DM in Slack. I only need it for testing and won't share it anywhere else.

I also left some suggestions regarding the metadata. 
Please check this document with guidelines: https://developers.raycast.com/basics/prepare-an-extension-for-store, it should be helpful with improving your extension.

File: extensions/youdao-translate/src/youdao_translate.tsx [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+        <SubmitFormAction title="Translate" onSubmit={(input: translateRequest) => {
+          push(
+            <Translate content={input.content || select} to_language={input.to_language}
+                       from_language={input.from_language} />
+          );
+        }} />
+      </ActionPanel>
+    }>
+      <Form.TextArea title="Content(内容)" id="content"
+                     placeholder={select !== "" ? `you selected ${select}` : "please enter content"} />
```
[the remark this answers is not in the document]
Comment (tiancheng92) [reply extensions/youdao-translate/src/youdao_translate.tsx]: When the user selects text, I want to display the selected text in the text area. If the user’s selected text is a misoperation, the user can overwrite the selected text with the entered text.

**Comment by xeric** [comment]
@PitNikola I created another youdao translation extension which focus on simplify the flow of user experience, it is using a list view and directly translate the selected word/clipboard content/input word with one step (no form view anymore), since I was heavily rely on a similar workflow in Alfred, I hope have a similar one in Raycast.
PR is here: https://github.com/raycast/extensions/pull/1004
@pernielsentikaer suggested we can collaborate and keep as one solution to end user, do you think it's possible?

**Comment by xeric** [comment]
> It's best to ask @tiancheng92 if he would be up for changes. If not (which is okay considering different approaches), it's okay to submit yours as separate one.

@tiancheng92 any idea about @PitNikola suggestion?

**Comment by PitNikola** [comment]
@xeric, since there is no response from @tiancheng92, I think you can go ahead with submitting separate extension with your own implementation. It's not ideal, but what can we do.

cc @pernielsentikaer

**Pull request merged**